src/field.cpp

```diff
@@ -0,0 +1,34 @@
+#include "field.h"
+
+typedef std::mt19937 rng_t;
+
+Field::Field(sf::RenderWindow& w):
+window(w)
+{
+    currY = 0;
+    distanceToFurthestObstacle = 0;
+    std::random_device ranDevice; // init random device
+    generator.seed(ranDevice()); // spawn generator with seed from ranDevice
+    udist = std::uniform_int_distribution<int>(0, FIELD_CONSTS::MAX_DIST_BETWEEN_OBSTACLES); // uniform distribution
+    
+}
+
+int Field::generateObstacle() 
+{
+    return distanceToFurthestObstacle + udist(generator);
```

Comment:
Is it ok if multiple obstacles have the same distance/position? If not u may want to account for this when generating a random number.

Reply:
It's fine.